src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public partial class DependencyContextJsonReader : IDependencyContextReader
+    {
+        public DependencyContext Read(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            ArraySegment<byte> drained = ReadToEnd(stream);
+
+            DependencyContext dependencyContext;
+            try
+            {
+                dependencyContext = ReadCore(drained.AsSpan(0, drained.Count));
+            }
+            catch
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+                throw;
+            }
+            finally
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+            }
+            return dependencyContext;
+        }
+
+        private static bool IsTokenTypeProperty(JsonTokenType tokenType)
+            => tokenType == JsonTokenType.PropertyName;
+
+        private DependencyContext ReadCore(ReadOnlySpan<byte> jsonData)
+        {
+            var reader = new Utf8JsonReader(jsonData, isFinalBlock: true, state: default);
+
+            reader.ReadStartObject();
+
+            string runtime = string.Empty;
+            string framework = string.Empty;
+            bool isPortable = true;
+            string runtimeTargetName = null;
+            string runtimeSignature = null;
+
+            CompilationOptions compilationOptions = null;
+            List<Target> targets = null;
+            Dictionary<string, LibraryStub> libraryStubs = null;
+            List<RuntimeFallbacks> runtimeFallbacks = null;
+
+            while (reader.Read() && IsTokenTypeProperty(reader.TokenType))
+            {
+                switch (reader.GetString())
```

Comment:
We need to materialize these strings?  We can't get back the span and do a comparison based on spans?

Reply:
I realize that this code can be optimized for performance/allocations but the intent was to keep the diff as minimal as possible as part of this port to highlight what is absolutely necessary.

I will file an issue to make optimizations and avoid materializing strings where possible, and defer performance or cosmetic changes to the future.